carp.common/src/commonMain/kotlin/dk/cachet/carp/common/application/devices/DefaultDeviceRegistration.kt

```diff
@@ -11,20 +12,26 @@ import kotlinx.serialization.Serializable
  * The base class can't be made concrete since this would prevent being able to serialize extending types (constructors may not contain parameters which are not properties).
  */
 @Serializable
-data class DefaultDeviceRegistration( override val deviceId: String ) : DeviceRegistration()
+data class DefaultDeviceRegistration(
+    @Required
+    override val deviceDisplayName: String? = null,
+    @Required
+    override val deviceId: String = UUID.randomUUID().toString()
```

Comment:
I think deviceId should come before display name. It would be more consistent with the other implementing classes

Reply:
Interesting you should notice this. I went back on forth on this. As in the other PR comment, I concluded when using `DefaultDeviceRegistration` in the majority of use cases you want a generated `UUID`.

In essence, this is really a placeholder type to be replaced with a more concrete `DeviceRegistration` once we get round to modeling that, e.g., including manufacturer name, OS version etc. in a `SmartphoneDeviceRegistration`. Perhaps, at that point, we can only come up with a non-generated `deviceId` for smartphones.

Which does remind me again of potential privacy issues and need to deanonymize `DeviceRegistration` for anonymous exports. 🤔

Reply:
oki doki